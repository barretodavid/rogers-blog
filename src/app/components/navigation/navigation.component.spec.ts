import { NavigationComponent } from './navigation.component';
import { TestBed } from '@angular/core/testing';
import { MatSidenavModule, MatToolbarModule, MatListModule, MatIconModule } from '@angular/material';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { RouterModule } from '@angular/router';
import { APP_BASE_HREF } from '@angular/common';

describe('NavigationComponent', () => {
  it('should render its template', () => {
    TestBed.configureTestingModule({
      imports: [
        NoopAnimationsModule,
        MatSidenavModule,
        MatToolbarModule,
        MatListModule,
        MatIconModule,
        RouterModule.forRoot([])
      ],
      declarations: [
        NavigationComponent,
      ],
      providers: [
        { provide: APP_BASE_HREF, useValue: '/' },
      ]
    }).compileComponents().then(() => {
      const fixture = TestBed.createComponent(NavigationComponent);
      fixture.detectChanges();
      expect(fixture).toMatchSnapshot();
    });
  });
});

