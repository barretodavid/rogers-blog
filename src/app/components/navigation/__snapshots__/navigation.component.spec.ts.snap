// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationComponent should render its template 1`] = `
<app-navigation
  blogTitle={[Function String]}
  breakpointObserver={[Function BreakpointObserver]}
  isHandset$={[Function Observable]}
>
  <mat-sidenav-container
    class="sidenav-container mat-drawer-container mat-sidenav-container"
  >
    
    <div
      class="mat-drawer-backdrop ng-star-inserted"
    />
    <div
      class="cdk-visually-hidden cdk-focus-trap-anchor"
    />
    <mat-sidenav
      class="sidenav mat-drawer mat-sidenav ng-tns-c1-0 ng-trigger ng-trigger-transform mat-drawer-side mat-sidenav-fixed ng-star-inserted ng-animating"
      fixedinviewport="true"
      ng-reflect-fixed-in-viewport="true"
      ng-reflect-mode="side"
      ng-reflect-opened="true"
      role="navigation"
      style="top:0px;bottom:0px;0:top;1:bottom;"
      tabindex="-1"
    >
      <div
        class="mat-drawer-inner-container"
      >
        <mat-toolbar
          class="mat-toolbar mat-toolbar-single-row"
        >
          Menu
        </mat-toolbar>
        <mat-nav-list
          class="mat-nav-list mat-list-base"
          role="navigation"
        >
          <a
            class="mat-list-item"
            href="/list"
            mat-list-item=""
            ng-reflect-router-link="list"
            routerlink="list"
          >
            <div
              class="mat-list-item-content"
            >
              <div
                class="mat-list-item-ripple mat-ripple"
                mat-ripple=""
                ng-reflect-disabled="false"
                ng-reflect-trigger="http://localhost/list"
              />
              <div
                class="mat-list-text"
              />
              Home
            </div>
          </a>
          <a
            class="mat-list-item"
            href="/create"
            mat-list-item=""
            ng-reflect-router-link="create"
            routerlink="create"
          >
            <div
              class="mat-list-item-content"
            >
              <div
                class="mat-list-item-ripple mat-ripple"
                mat-ripple=""
                ng-reflect-disabled="false"
                ng-reflect-trigger="http://localhost/create"
              />
              <div
                class="mat-list-text"
              />
              Create Post
            </div>
          </a>
          <a
            class="mat-list-item"
            href="/contact"
            mat-list-item=""
            ng-reflect-router-link="contact"
            routerlink="contact"
          >
            <div
              class="mat-list-item-content"
            >
              <div
                class="mat-list-item-ripple mat-ripple"
                mat-ripple=""
                ng-reflect-disabled="false"
                ng-reflect-trigger="http://localhost/contact"
              />
              <div
                class="mat-list-text"
              />
              Contact
            </div>
          </a>
        </mat-nav-list>
      </div>
    </mat-sidenav>
    <div
      class="cdk-visually-hidden cdk-focus-trap-anchor"
    />
    <mat-sidenav-content
      class="mat-drawer-content mat-sidenav-content"
    >
      <mat-toolbar
        class="mat-toolbar mat-primary mat-toolbar-single-row"
        color="primary"
        ng-reflect-color="primary"
      >
        
        <span>
          Rogers Blog
        </span>
      </mat-toolbar>
      <router-outlet />
    </mat-sidenav-content>
    
  </mat-sidenav-container>
</app-navigation>
`;
